data/data/azure/bootstrap/main.tf

```diff
@@ -90,15 +102,10 @@ resource "azurerm_network_interface" "bootstrap" {
     subnet_id                     = var.subnet_id
     name                          = local.bootstrap_nic_ip_configuration_name
     private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = azurerm_public_ip.bootstrap_public_ip.name
```

Comment:
nit: this must be `azurerm_public_ip.bootstrap_public_ip.id`

Reply:
Fixed!